src/components/modal/personalization/paginated/personalization-paginated.modal.tsx

```diff
@@ -0,0 +1,125 @@
+import { useState } from 'react';
+import { Grid } from 'theme-ui';
+import {
+    Pill,
+    Button,
+    Checkbox,
+    Pagination,
+    TypographyScale,
+    ESystemIconNames,
+} from '@mdb/flora';
+import { TopicCard } from '@mdb/devcenter-components';
+
+import { useModalContext } from '../../../../contexts/modal';
+
+import { Tag } from '../../../../interfaces/tag';
+import {
+    initializePersonalizationConfig,
+    submitPersonalizationSelections,
+} from '../utils';
+import { tagToTopic } from '../../../../utils/tag-to-topic';
+
+import styles from '../styles';
+
+const PaginatedPersonalizationModal = () => {
+    const paginationConfig = initializePersonalizationConfig();
```

Comment:
can this be completely moved outside of the render loop, since it doesn't depend on any props or state? if i'm reading the code correctly, this is going to cause a ~100 iteration loop to run every time the component re-renders (ie any time a user interacts with the modal). i think maybe `initializePersonalizationConfig` can just be an exported array that's imported in this file, or we can consider even converting it into a preval'd value.

Reply:
same here as well https://github.com/mongodb/devcenter/pull/414/files#diff-e7bda44f51bf7eb9974a43ecda03021180955a551dd45f7a97efd110c72b4e10R24

Reply:
> I don't know exactly why, but these changes seem to cause what's almost a full page re-render whenever a modal is opened or closed, which adds some lag/lack of responsiveness.
> 
> Before when opening a modal: ![Screen Shot 2023-01-25 at 12 11 45 PM](https://user-images.githubusercontent.com/110849018/214633119-38ea0cc6-1bb9-4b6c-bea8-3d391c215e0e.png)
> 
> After: ![Screen Shot 2023-01-25 at 12 11 40 PM](https://user-images.githubusercontent.com/110849018/214633196-bc10b21a-ca83-43ed-ad32-6da8fe7bc6d7.png)
> 
> This doesn't make much sense to me because context was being used before as well. Maybe not too much of an issue for now, but I thought I'd flag it in case we want to look into it in the future.
> 
> EDIT: Did some digging and it has to do with [this line](https://github.com/mongodb/devcenter/pull/414/files#diff-0b75a532e530ce749a966d77dffeea3593e67560e5894213143637ee1d84aef5R57). Honestly I'm probably missing something but I've looked at the modals/overlays with and without this overflow switching and I can't tell a difference. Can this just be removed and `globalStyles` converted to a string instead of a function?

You're seeing the difference in the perf probably because the existing modals didn't try to set overflow: hidden and would just let the window scroll while the modal was open. I wasn't told this was _necessarily_ a defect, I just think its bad UX so I added the overflow hidden when all modals are open, especially because both personalization modals will have inner scroll, so its going to cause some wonky behavior if the window also has scroll while the user is trying to only scroll the modal

Reply:
can look into a way getting a way around this without calling functions in styled if its a big issue, but I do think its something we're going to need to have

Reply:
gotcha, as far as the global styling I think we can just figure it out later, not too big of a deal to have half a second lag only when opening/closing modals